### ISSUE TITLE
Keyword replacement logic doesn't preserve items within `guardianFactorProviders`

### ISSUE THREAD

**Issue by brendenpalmer** [auth0/auth0-deploy-cli#845]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/auth0-deploy-cli#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://github.com/auth0/auth0-deploy-cli#documentation) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-deploy-cli/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/auth0-deploy-cli/releases/latest) of this tool and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/tag/auth0-deploy-cli) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Keywords within `guardianFactorProviders` don't seem to be preserved as expected on export. The default config for `guardianFactorProviders` is here: [src/tools/auth0/handlers/guardianFactorProviders.ts#L34](https://github.com/auth0/auth0-deploy-cli/blob/master/src/tools/auth0/handlers/guardianFactorProviders.ts#L34). It sets `id` to `name`, which conflicts with logic here: [src/keywordPreservation.ts#L235](https://github.com/auth0/auth0-deploy-cli/blob/master/src/keywordPreservation.ts#L235) and its usage here: [src/keywordPreservation.ts#L56](https://github.com/auth0/auth0-deploy-cli/blob/master/src/keywordPreservation.ts#L56); as a result, it'll attempt to match based on `id`, but `id` won't exist in the `tenant.yml` file so the keyword will ultimately never be preserved.

### Expectation

Should have keywords preserved within `guardianFactorProviders`

### Reproduction

See failing test-case here: https://github.com/auth0/auth0-deploy-cli/compare/master...brendenpalmer:auth0-deploy-cli:failing-test-case-guardian-factor-providers?expand=1

### Deploy CLI version

7.19.0

### Node version

19.7.0

**Comment by willvedd** [comment]
@brendenpalmer Thanks for opening this ticket, especially appreciative of the unit test. This is definitely a bug that impacts all of the guardian resource types. Will need to dig into this a bit further but think we can address it sometime in the near future.

**Comment by brendenpalmer** [comment]
@willvedd I wonder if an option could be to not filter out the handler `id` when constructing `resourceSpecificIdentifiers` and instead match on all identifiers, but skip keys not present on the resource when constructing the specific address. Potentially something like: https://github.com/auth0/auth0-deploy-cli/commit/8a8696e9e8ec85fb2d07cfc8cd7eada11d314056

**Comment by esumerfd** [comment]
This may be related. I am attempting to disable guardian factors sms.json. If true updates but when false it does nothing.
```
{
  "name": "sms",
  "enabled": ##ENABLE_SMS##
}
```

**Comment by willvedd** [comment]
@esumerfd I believe you're referring to a separate issue. I'd leverage the `@@` literal-replacement syntax instead of the `##` string-replacement syntax. This _should_ achieve your desired result. Although, on subsequent export, that replacement marker will get blown away due to OP's issue. 

**Example:**

```
{
  "name": "sms",
  "enabled": "@@ENABLE_SMS@@"
}
```

**Comment by mrsarm** [comment]
I was having the same issue in a project also with the version 7.19.0, and downgrading to 7.18.0 solved the problem for me, although there are other tickets reported in this project by people having the same issue with the older version too.

The `audience` placeholder was replaced by the actual value during the export with the newest version:

**tenant.yaml:**

```yaml
clientGrants:
  - ...
  - client_id: Portal App
    audience: '##API_AUDIENCE##'
    scope:
      - read:users
      - update:users
      - ...
```

**config.stage.json:**

```json
{
  ...
  "AUTH0_ALLOW_DELETE": true,
  "AUTH0_PRESERVE_KEYWORDS": true,
  "AUTH0_KEYWORD_REPLACE_MAPPINGS": {
    "API_AUDIENCE": "https://some-stage-domain.auth0.com/api/v2/",
    "ALLOWED_LOGOUT_URLS": [
      ...
    ],
    "CALLBACKS": [
      ...
    ]
  }
}
```

Command executed: `a0deploy export --format=yaml -c config.stage.json --output_folder=resources`.

**Comment by willvedd** [comment]
@mrsarm You're describing a similar but separate issue that's discussed in #855. Please see my comment about it. AFAIK, simply downgrading will not fix the issue.

**Comment by willvedd** [comment]
This specific keyword preservation issue has been fixed with [7.20.0](https://github.com/auth0/auth0-deploy-cli/releases/tag/v7.20.0). It should improve the feature's coverage to not just `guardianFactorProviders` but for many more resources. Appreciate everyone's feedback here.

**Comment by rakeshpawar09** [comment]
7.24.2 and 7.24.1 have the same issue , audience param is not getting preserved during export.
